v-next/hardhat-node-test-reporter/src/error-formatting.ts

```diff
@@ -96,10 +97,27 @@ function formatSingleError(
 
   const diff = getErrorDiff(error);
 
-  const stackReferences: StackReference[] = stackLines
+  let stackReferences: StackReference[] = stackLines
     .map(parseStackLine)
     .filter((reference) => reference !== undefined);
 
+  // Remove all the stack references exposing test runner internals
+  // User code starts after: (Suite|Test|TestHook|...).runInAsyncScope
+  const runInAsyncScopeIndex = stackReferences.findIndex((reference) => {
+    return (
+      reference.context !== undefined &&
+      reference.context.endsWith(".runInAsyncScope")
+    );
+  });
+  if (runInAsyncScopeIndex !== -1) {
+    stackReferences = stackReferences.slice(0, runInAsyncScopeIndex);
+  }
+
+  // Remove all the stack references originating from node
```

Comment:
I think we should probably remove all the bottom ones, instead of all of them.

Imagine something like this:

```
- node:http/something/something.js:123
- myFileThatDoesAnHttpRequest.js:345
- node:something/else.js:78
```

Id't like to see 

```
- node:http/something/something.js:123
- myFileThatDoesAnHttpRequest.js:345
```

and not just

```
- myFileThatDoesAnHttpRequest.js:345
```

as the exception isn't coming from my code.

Reply:
What if all the frames come from node? That'd be some sort of internal error, and in that case we probably don't want to remove anything. Maybe not even the test runner ones.

Reply:
That's a great point. I'm going to update the proposed implementation to remove only continuously grouped node references, and not to remove them at all if there are only node references in the stack.